packages/lib/src/vis/heatmap/HeatmapMaterial.tsx

```diff
@@ -0,0 +1,153 @@
+    alphaValues,
+    alphaDomain = DEFAULT_DOMAIN,
+  } = props;
+
+  const dataTexture = useMemo(
+    () => getDataTexture(values, magFilter),
+    [magFilter, values]
+  );
+
+  const alphaTexture = useMemo(
+    () => alphaValues && getDataTexture(alphaValues),
+    [alphaValues]
+  );
+
+  const colorMapTexture = useMemo(() => {
+    const interpolator = getInterpolator(colorMap, invertColorMap);
+
+    const colors = Uint8Array.from(
+      Array.from({ length: CMAP_SIZE }).flatMap((_, i) => {
+        const { r, g, b } = rgb(interpolator(i / (CMAP_SIZE - 1)));
+        return [r, g, b, 255];
+      })
+    );
+
+    const texture = new DataTexture(
+      colors,
+      CMAP_SIZE,
+      1,
+      RGBAFormat,
+      UnsignedByteType
+    );
+    texture.needsUpdate = true;
+
+    return texture;
+  }, [colorMap, invertColorMap]);
+
+  const [scaleType, gammaExponent] = Array.isArray(visScaleType)
+    ? visScaleType
+    : [visScaleType as ScaleType, 1];
+
+  const scaledDomain = scaleDomain(domain, scaleType);
+
+  const shader = {
+    uniforms: getUniforms({
+      data: dataTexture,
+      colorMap: colorMapTexture,
+      min: scaledDomain[0],
+      oneOverRange: 1 / (scaledDomain[1] - scaledDomain[0]),
+      gammaExponent,
+      normRevertFactor: values.dtype === 'uint8' ? 255 : 1, // revert WebGL's automatic normalization of UNSIGNED_BYTE with RED format
+      alpha: alphaTexture,
+      withAlpha: alphaValues ? 1 : 0,
+      alphaMin: alphaDomain[0],
+      oneOverAlphaRange: 1 / (alphaDomain[1] - alphaDomain[0]),
+    }),
+    vertexShader: VERTEX_SHADER,
+    fragmentShader: `
+      uniform sampler2D data;
+      uniform sampler2D colorMap;
+
+      uniform float min;
+      uniform float oneOverRange;
+      uniform float gammaExponent;
+      uniform float normRevertFactor;
+
+      uniform sampler2D alpha;
+      uniform float alphaMin;
+      uniform float oneOverAlphaRange;
+      uniform int withAlpha;
+
+      const vec4 nanColor = vec4(255, 255, 255, 1);
+      const float oneOverLog10 = 0.43429448190325176;
+
+      varying vec2 coords;
+
+      bool isSupported(float value) {
+        return ${IMPL_IS_SUPPORTED[scaleType] || 'true'};
+      }
+
+      float scale(float value) {
+        return ${IMPL_SCALE[scaleType] || 'value'};
+      }
+
+      void main() {
+        float value = texture2D(data, coords).r * normRevertFactor;
+
+        if (isnan(value) || isinf(value) || !isSupported(value)) {
+          gl_FragColor = nanColor;
+        } else {
+          float scaledValue = scale(value);
+          float normalizedValue = clamp(oneOverRange * (scaledValue - min), 0., 1.);
+
+          gl_FragColor = texture2D(colorMap, vec2(pow(normalizedValue, gammaExponent), 0.5));
+
+          if (withAlpha == 1) {
+            gl_FragColor.a = oneOverAlphaRange * (texture2D(alpha, coords).r - alphaMin);
+          }
+        }
+      }
+    `,
+  };
+
+  return <shaderMaterial args={[shader]} side={DoubleSide} />;
```

Comment:
Note the addition of `side` that changes nothing for `HeatmapMesh` but makes the material visible from both sides in 3D